nunit/MVP-Advance/SeleniumNUnit/Pages/ManageListings.cs

```diff
@@ -36,9 +42,93 @@ internal class ManageListings
         private IWebElement btnSave => driver.FindElement(By.XPath("//input[@value='Save']"));
         #endregion
 
-        internal void AddListing()
+        ShareSkill shareSkillObj;
+
+        //Add a skill
+        internal void AddListing(int rowNumber, string worksheet)
+        {
+            shareSkillObj = new ShareSkill();
+            btnShareSkill.Click();
+            wait(2);
+            shareSkillObj.EnterShareSkill(rowNumber, worksheet);
+            wait(3);
+        }
+
+        //Verify add & edit
+        internal void ViewListing(int rowNumber, string worksheet)
+        {
+
+            //Click on ManageListing
+            GoToManageListings();
+            WaitForElement(driver, By.XPath(eTable), 5);
+
+            //Read data
+            ExcelLib.PopulateInCollection(Base.ExcelPath, worksheet);
+            string expectedTitle = ExcelLib.ReadData(rowNumber, "Title");
+
+            //Click on button View
+            string e_View = "//div[@id='listing-management-section']//tbody/tr[" + GetTitleIndex(expectedTitle) + "]/td[8]/div/button[1]";
```

Comment:
any reason this is here and not as a page object?

Reply:
Yes. It's a dynamic string of an element where I need to concatenate a prefix, an integer, and a suffix. The integer is the return value from my GetTitleIndex method that looks for the position of a  skill title in the listing. The dynamic string then combines into a complete element of the button view of the matching title.
I tried writing in the page object but then gave me some errors, because it references another method.